src/promptflow-core/promptflow/core/_serving/app.py

```diff
@@ -2,281 +2,48 @@
-            connections_name_overrides=self.connections_name_override,
-            # for serving, we don't need to persist intermediate result, this is to avoid memory leak.
-            storage=DummyRunStorage(),
-            credential=self.credential,
-            init_kwargs=self.init,
-        )
-        # why we need to update bonded executable flow?
-        self.flow = self.flow_invoker.flow
-        # Set the flow name as folder name
-        self.flow.name = self.flow_name
-        self.response_fields_to_remove = get_output_fields_to_remove(self.flow, logger)
-        logger.info("Promptflow executor initializing succeed!")
-
-    def init_swagger(self):
-        self.response_fields_to_remove = get_output_fields_to_remove(self.flow, logger)
-        self.swagger = generate_swagger(self.flow, self.sample, self.response_fields_to_remove)
-        data = load_feedback_swagger()
-        self.swagger["paths"]["/feedback"] = data
+from promptflow.core._serving.v1.app import PromptflowServingApp
+from promptflow.core._serving.v2.app import PromptFlowServingAppV2
 
-    def set_default_environment_variables(self, default_environment_variables: Dict[str, str] = None):
-        if default_environment_variables is None:
-            return
-        for key, value in default_environment_variables.items():
-            if key not in os.environ:
-                os.environ[key] = value
 
-
-def add_default_routes(app: PromptflowServingApp):
-    @app.errorhandler(Exception)
-    def handle_error(e):
-        err_resp, resp_code = handle_error_to_response(e, logger)
-        app.flow_monitor.handle_error(e, resp_code)
-        return err_resp, resp_code
-
-    @app.route("/score", methods=["POST"])
-    @enable_monitoring
-    def score():
-        """process a flow request in the runtime."""
-        raw_data = request.get_data()
-        logger.debug(f"PromptFlow executor received data: {raw_data}")
-        app.init_invoker_if_not_exist()
-        if app.flow.inputs.keys().__len__() == 0:
-            data = {}
-            logger.info("Flow has no input, request data will be ignored.")
-        else:
-            logger.info("Start loading request data...")
-            data = load_request_data(app.flow, raw_data, logger)
-        # set context data
-        g.data = data
-        g.flow_id = app.flow.id or app.flow.name
-        run_id = g.get("req_id", None)
-        # TODO: refine this once we can directly set the input/output log level to DEBUG in flow_invoker.
-        disable_data_logging = logger.level >= logging.INFO
-        span = trace.get_current_span()
-        if span == INVALID_SPAN:
-            # no parent span, try to extract trace context from request
-            ctx = try_extract_trace_context(logger)
-        else:
-            ctx = None
-        token = context.attach(ctx) if ctx else None
+def create_app(**kwargs):
+    engine = kwargs.pop("engine", "flask")
+    if engine == "flask":
```

Comment:
User current not setting this engine. Do we expect to switch to this by default? Or user does that in their Dockerfile?

Shall we add doc for this new engine and when to use this?

Reply:
the default value currently is still flask,  for short term we will provide environment variable for customer to choose switching to use fastapi,  once we have enough confidence we plan to switch to use fastapi as default.

As for doc, definitely we will update the doc for  how to enable this.